### ISSUE TITLE
[CVP] Missed optimization for smax((a + 7) / 8, 1) where a s> 0

### ISSUE THREAD

**Issue by XChy** [llvm/llvm-project#72378]
Alive2 proof: https://alive2.llvm.org/ce/z/Cyoo2Y
Missed example: https://godbolt.org/z/9zW8PToPe

Real-world motivation: This snippet of IR is derived from FFmpeg/.../extr_aacdec.c_latm_decode_audio_specific_config.c (after O3 pipeline)
The example above is a reduced version. If you're interested in the original suboptimal IR and optimal IR. See also: https://godbolt.org/z/4a6K4b8hc

**Let me know if you can confirm that it's an optimization opportunity, thanks.**

**Comment by antoniofrighetto** [comment]
Looks a valid optimization to me. Not sure if, in principle, this should live in CVP or somewhere in ConstantPropagation.

**Comment by XChy** [comment]
> Looks a valid optimization to me. Not sure if, in principle, this should live in CVP or somewhere in ConstantPropagation.

@antoniofrighetto thanks for feedback! I've changed the title until the pass to blame is specified.

**Comment by nikic** [comment]
@XChy I don't think that would be sufficient here. The issue is that `%div` is defined outside the branch condition and will use the value of `%a` without the condition. Even if you use AtUse API, it will not be able to constrain the result of `%div` (as the condition is on `%a`, not `%div`).

**Comment by XChy** [comment]
@nikic You're right, I missed that we can constrain only one value at a time now. 
For one-use example in this issue, a easy solution is to sink these one-use instructions as `SinkPass` does. But for multi-use scenario in real world, that's a bit tricky to handle.
